src/analyses/variable-sensitivity/abstract_enviroment.cpp

```diff
@@ -0,0 +1,585 @@
+        const auto &symbol_entry=map.find(symbol);
+        if(symbol_entry==map.cend())
+        {
+          return abstract_object_factory(expr.type(), true);
+        }
+        else
+        {
+          return symbol_entry->second;
+        }
+      }
+    },
+    {
+      ID_constant,[&](const exprt &expr)
+      {
+        return abstract_object_factory(
+          expr.type(), to_constant_expr(expr));
+      }
+    },
+    {
+      ID_member, [&](const exprt &expr)
+      {
+        member_exprt member_expr(to_member_expr(expr));
+        sharing_ptrt<struct_abstract_objectt> struct_abstract_object=
+          std::dynamic_pointer_cast<struct_abstract_objectt>(
+            eval(member_expr.compound(), ns));
+
+        return struct_abstract_object->read_component(*this, member_expr);
+      }
+    },
+    {
+      ID_address_of, [&](const exprt &expr)
+      {
+#if 0
+        address_of_exprt address_expr(to_address_of_expr(expr));
+#endif
+        // TODO(tkiley): This needs special handling
+        // For now just return top
+        return abstract_object_factory(expr.type(), true, false);
+      }
+    },
+    {
+      ID_dereference, [&](const exprt &expr)
+      {
+        dereference_exprt dereference(to_dereference_expr(expr));
+        sharing_ptrt<pointer_abstract_objectt> pointer_abstract_object=
+          std::dynamic_pointer_cast<pointer_abstract_objectt>(
+            eval(dereference.pointer(), ns));
+
+        return pointer_abstract_object->read_dereference(*this);
+      }
+    },
+    {
+      ID_index, [&](const exprt &expr)
+      {
+        index_exprt index_expr(to_index_expr(expr));
+        sharing_ptrt<array_abstract_objectt> array_abstract_object=
+          std::dynamic_pointer_cast<array_abstract_objectt>(
+            eval(index_expr.array(), ns));
+
+        return array_abstract_object->read_index(*this, index_expr);
+      }
+    }
+  };
+
+  // first try to collapse constant folding
+  const exprt &simplified_expr=simplify_expr(expr, ns);
+
+  const auto &handler=handlers.find(simplified_expr.id());
+  if(handler==handlers.cend())
+  {
+    // No special handling required by the abstract environment
+    // delegate to the abstract object
+    if(simplified_expr.operands().size()>0)
+    {
+      return eval_expression(simplified_expr, ns);
+    }
+    else
+    {
+      return abstract_object_factory(simplified_expr.type(), true);
+    }
+  }
+  else
+  {
+    return handler->second(simplified_expr);
+  }
+}
+
+/*******************************************************************\
+
+Function: abstract_environmentt::assign
+
+  Inputs:
+   expr - the expression to assign to
+   value - the value to assign to the expression
+
+ Outputs: ?
+
+ Purpose: Assign a value to an expression
+
+\*******************************************************************/
+
+bool abstract_environmentt::assign(
+  const exprt &expr, const abstract_object_pointert value)
+{
+  exprt s = expr;
```

Comment:
Throughout this function (and possibly the entire PR): no space around `=`, `==`, `!=`